### PR TITLE
Added artifact feature to pylint yaml

### LINKED ISSUE #264: Add output artifact to the pylint yaml.
As a developer, when I create new pull requests to the repository, I want a summary to be produced of the pylint results to securely document my work.

### PR DESCRIPTION
**Pull request by olliejth**
Pylint data should now be uploaded as an artifact.

closes #264

### CODE REVIEW

File: .github/workflows/py_lint.yml [hunk 1]
```diff
@@ -31,4 +31,10 @@ jobs:
           find . -name 'requirements.txt' -exec pip install -r {} \;
 
       - name: Run pylint
-        run: pylint **/*.py --fail-under=8
+        run: pylint **/*.py --fail-under=8 > pylint_report.txt
+
+      - name: Upload pylint report as artifact
+        uses: actions/upload-artifact@v3
```
Comment (AlexOBarnes) [inline .github/workflows/py_lint.yml:37]: Looks good

**Review by AlexOBarnes — APPROVED** [review]
Looks good to me

File: .github/workflows/py_lint.yml [hunk 2]
```diff
@@ -31,4 +31,10 @@ jobs:
           find . -name 'requirements.txt' -exec pip install -r {} \;
 
       - name: Run pylint
-        run: pylint **/*.py --fail-under=8
+        run: pylint **/*.py --fail-under=8 > pylint_report.txt
```
Comment (CalebMathers) [inline .github/workflows/py_lint.yml:34]: Good to get this in a report we can access

File: .github/workflows/py_lint.yml [hunk 3]
```diff
@@ -31,4 +31,10 @@ jobs:
...
+
+      - name: Upload pylint report as artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: pylint-report
```
Comment (CalebMathers) [inline .github/workflows/py_lint.yml:39]: LGTM

**Review by CalebMathers — APPROVED** [review]

**Pull request merged**